### PR TITLE
feat: add `/usd-order` route for TGE

### PR DESCRIPTION
**Pull request by tcoratger**
## Description
Adds a new route `/usd-order` to facilitate USD-based orders for TGE.

## Changes Made
- Implemented the `/usd-order` route in the TGE module to handle USD-based orders for Gx tokens.
- Unit tests for the route
- Improve naming and doc

### CODE REVIEW

File: src/routes/tge.ts [hunk 1]
```diff
@@ -232,14 +243,201 @@ router.post('/pre-order', authenticateApiKey, async (req, res) => {
...
+/**
+ * POST /v1/tge/usd-order
+ *
+ * @summary Process USD-based Gx token order
+ * @description Processes an order for Gx tokens using USD, linking it to the provided quote ID and user details.
+ * @tags USD Order
+ * @security BearerAuth
+ * @param {string} req.body.quoteId - The quote ID to process the order.
+ * @param {string} req.body.userTelegramID - The user's Telegram ID for identification.
+ * @param {string} req.body.token_address - The token address for USD-based order.
```
Comment (milukove) [inline src/routes/tge.ts:268]: @tcoratger  Why `token_address` suddenly? I thought we are using camelCase.

**Review by milukove — APPROVED** [review]
@tcoratger  Looks good, thank you. 

Just want to suggest some adjustments to the endpoint paths:

```
GET /v1/tge/conversion-information => GET /v1/tge/quote
POST /v1/tge/pre-order => POST /v1/tge/order
POST /v1/tge/usd-order => PATCH /v1/tge/order/:orderId
GET /v1/tge/orders is good
GET /v1/tge/quotes is good
GET /v1/tge/order-status => GET /v1/tge/order/:orderId
```

WDT?

File: src/routes/tge.ts (hunk 1 above)
Comment (tcoratger) [reply src/routes/tge.ts]: Correct just changed :)

**Comment by tcoratger** [comment]
> @tcoratger Looks good, thank you.
> 
> Just want to propose some adjustments to the endpoint paths:
> 
> ```
> GET /v1/tge/conversion-information => GET /v1/tge/quote
> POST /v1/tge/pre-order => POST /v1/tge/order
> POST /v1/tge/usd-order => PATCH /v1/tge/order/:orderId
> GET /v1/tge/orders is good
> GET /v1/tge/quotes is good
> GET /v1/tge/order-status => GET /v1/tge/order/:orderId
> ```
> 
> WDT?

@milukove Correct just did the modifications. Small clarification on the PATCH method, I kept the entire payload in a body because there is not only the order ID, there is also the Telegram ID of the user which is provided to ensure that the user is the correct one. As well as the address of the token and the chain ID. So I think it's better to leave it all in a body, WDT?

**Comment by milukove** [comment]
@tcoratger Looks good, thank you.

**Pull request merged**